src/plots/sort_modules.js

```diff
@@ -0,0 +1,33 @@
+/**
+* Copyright 2012-2018, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+function sortBasePlotModules(a, b) {
+    var nameA = a.name;
+    var nameB = b.name;
+
+    // always plot splom before cartesian (i.e. scattergl traces)
+    if(nameB === 'splom' && nameA === 'cartesian') {
+        return 1;
+    }
```

Comment:
I don't think this is sufficient... if there are more than those two modules you're not even guaranteed that `sort` will ever be called with that exact pair; and if it is called with that pair, you're not guaranteed that they'll be in that order, nor that [`0` will preserve order](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort).

Seems to me you need to either put `splom` before everything, or `cartesian` after everything, eg:
```js
if(nameB === 'splom') return 1;
if(nameA === 'splom') return -1;
return 0;
```

Reply:
thanks for the tip!

Reply:
fixed in https://github.com/plotly/plotly.js/pull/2505/commits/31e34fac459e22ececca32bdc4583378741b2780